src/tools/qt-gui/src/treeviewmodel.cpp

```diff
@@ -489,43 +498,26 @@ void TreeViewModel::populateModel ()
 	populateModel (config);
 }
 
+void TreeViewModel::addRootNode (ElektraNamespace ns)
+{
+	kdb::Key k;
+	k.setNamespace (ns);
+
+	auto name = k.getName ();
+	name = name.substr (0, name.length () - 1);
+
+	m_model << ConfigNodePtr (new ConfigNode (QString::fromStdString (std::string{ static_cast<char> (ns) }),
+						  QString::fromStdString (name), nullptr, this, true));
+}
+
 void TreeViewModel::populateModel (KeySet const & keySet)
 {
 	m_model.clear ();
 
-	using namespace ckdb; // for namespaces
-	for (int i = KEY_NS_FIRST; i <= KEY_NS_LAST; ++i)
-	{
-		elektraNamespace ns = static_cast<elektraNamespace> (i);
-		ConfigNodePtr toAdd;
-		switch (ns)
-		{
-		case KEY_NS_SPEC:
-			toAdd = ConfigNodePtr (new ConfigNode ("spec", "spec", nullptr, this));
-			break;
-		case KEY_NS_PROC:
-			// TODO: add generic commandline parsing
-			break;
-		case KEY_NS_DIR:
-			toAdd = ConfigNodePtr (new ConfigNode ("dir", "dir", nullptr, this));
-			break;
-		case KEY_NS_USER:
-			toAdd = ConfigNodePtr (new ConfigNode ("user", "user", nullptr, this));
```

Comment:
Simply adding `:` here did not help?

Reply:
No, because we use the C++ key name iterator. That iterates over unescaped key names, so we need `std::string{ static_cast<char> (ns) }` to match correctly. In other words the `ConfigNode::getName` should return an unescaped key name part.

What I don't know is, if the `name` of the root `ConfigNode`s is ever checked outside of for loop in `createNewNodes`. If not there may have been easier solutions.

Reply:
The root node is only there to be the parent of all other nodes, AFAIR it is never touched again.